src/foam/nanos/auth/HistoricPassword.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'HistoricPassword',
+
+  documentation: 'Historic hashed password value for a user.',
+
+  javaImports: [
+    'java.util.Date'
+  ],
+
+  properties: [
+    {
+      class: 'DateTime',
+      name: 'timeStamp',
+      visibility: 'RO',
```

Comment:
`visibility` is deprecated. Please use `createMode`, `readMode`, and `updateMode` instead. In this case, you'd just want `updateMode: 'RO'`.